### ISSUE TITLE
fix: tidy up release steps

### ISSUE THREAD

**Issue by SgtPooki** [ipfs/ipfs-desktop#2470]
We should tidy up our release steps so that automating ipfs-desktop release will be more simple. 

Automating changelog creation saves time and ensures accuracy. It's also important to include "docs" commits in each release, so changes are well-documented and easy to find. These simple steps can make the release process smoother and more efficient.

Some existing efforts pushed to https://github.com/ipfs/ipfs-desktop/tree/feat/changelog-automation

```[tasklist]
### Tasks
- [ ] Automate changelog creation with semantic release
- [ ] "npm run release" runs steps 1, 2, 3, 4, & 7 from https://github.com/ipfs/ipfs-desktop/blob/main/DEVELOPER-NOTES.md#main-release-process automatically
```

**Comment by SgtPooki** [comment]
This was a small piece i wanted to write out that is going to be needed regardless of release please.

**Comment by lidel** [comment]
Both tasks are covered by #2664 (v0.32.0). release-please generates the release PR and the changelog from conventional commits, and CI cuts the GitHub release when that PR merges, so the numbered steps this issue wanted wrapped into `npm run release` no longer exist in DEVELOPER-NOTES.md.

Two steps stay manual on purpose. The release is created as a draft (`"draft": true` in `release-please-config.json`), so it is marked latest and published only after the manual test checklist. And `brew bump --open-pr homebrew/cask/ipfs` opens a PR in someone else's repo. Both are written up in DEVELOPER-NOTES.md.

Closing. If either of those should be automated as well, that is worth its own issue.
